components/Search.js

```diff
@@ -0,0 +1,56 @@
+class Search {
+  constructor(app) {
+    this.app = app;
+    this.searchInput = document.querySelector(".search-input");
+    this.searchButton = document.querySelector(".search-button");
+
+    this.setupEventListeners();
+  }
+
+  setupEventListeners() {
+    this.searchButton.addEventListener("click", () => {
+      const city = this.searchInput.value.trim();
+      if (city) {
+        this.fetchCityCoordinates(city);
+      } else {
+        alert("Please enter a city name!");
+      }
+    });
+  }
+
+  async fetchCityCoordinates(city) {
+    const apiKey = "50a738f4ae0fdc05df88af4f6a0dcf5f";
+    const url = `https://api.openweathermap.org/geo/1.0/direct?q=${city}&limit=1&appid=${apiKey}`;
+
+    try {
+      const response = await fetch(url);
+      if (!response.ok) {
+        throw new Error("Error fetching city coordinates");
+      }
+
+      const data = await response.json();
+      if (data.length === 0) {
+        alert("City not found. Please try again!");
+        return;
+      }
+
+      const { lat, lon } = data[0];
+      this.app.location = { latitude: lat, longitude: lon };
+
+      document.querySelector(
+        ".latitude-text"
+      ).textContent = `latitude: ${lat.toFixed(2)}°`;
+      document.querySelector(
+        ".longitude-text"
+      ).textContent = `longitude: ${lon.toFixed(2)}°`;
+
+      this.app.fetchWeather(lat, lon);
+      this.app.mapDisplay.getCountryAndCity(lat, lon);
+    } catch (error) {
+      console.error("Error fetching city coordinates:", error);
+      alert("Failed to fetch city coordinates. Please try again later.");
+    }
+  }
```

Comment:
move fetching logic to searchService.

Reply:
https://github.com/VasiliiDubik/Vasiliidubik-weatherForecast/pull/1/commits/8c6053dfabf871df1ad1226e4b6fccef4abf3565
services/searchService.js